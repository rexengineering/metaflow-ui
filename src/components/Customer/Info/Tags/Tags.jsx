import React from "react";
import PropTypes from "prop-types";
import InfoSection from "../InfoSection";
import ActionItem from "../../../ActionItem";
import { ActionItemDisplayOnlyType } from "../../../../constants";

function Tags({ tagsItems, className, onAddTagsButtonClicked }) {
  return (
    <InfoSection
      className={className}
      onAddIconClicked={onAddTagsButtonClicked}
      title="Tags"
    >
      {Array.isArray(tagsItems) &&
        tagsItems.map((item) => (
          <ActionItem
            data-testid="action-item"
            key={item}
            label={item}
            actionType={ActionItemDisplayOnlyType}
          />
        ))}
    </InfoSection>
  );
}

Tags.propTypes = {
  tagsItems: PropTypes.arrayOf(PropTypes.string).isRequired,
  onAddTagsButtonClicked: PropTypes.func.isRequired,
  className: PropTypes.string,
};

Tags.defaultProps = {
  className: "",
};

export default Tags;
