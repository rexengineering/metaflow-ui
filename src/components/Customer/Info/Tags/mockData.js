export default ["Direct Buyer", "RHL"];
