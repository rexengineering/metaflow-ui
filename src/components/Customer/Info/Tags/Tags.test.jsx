import { render, screen, fireEvent } from "@testing-library/react";
import React from "react";
import Tags from "./Tags";
import tagsItems from "./mockData";

describe("<TagsInfo />", () => {
  const renderTagsComponent = (onClick = undefined, tags = tagsItems) => {
    return (
      <Tags
        tagsItems={tags}
        onAddTagsButtonClicked={onClick}
        className="name"
      />
    );
  };

  it("should render", () => {
    const { asFragment } = render(renderTagsComponent());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(renderTagsComponent());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });

  it("should render all passed tag items", () => {
    render(renderTagsComponent());
    tagsItems.forEach((tag) => {
      const tagNode = screen.getByText(tag);
      expect(tagNode).toBeInTheDocument();
    });
  });

  it("should trigger the passed onClick function when clicked", () => {
    const onClick = jest.fn();
    render(renderTagsComponent(onClick));
    const element = screen.getByRole("button");
    fireEvent.click(element);

    expect(onClick).toHaveBeenCalled();
  });
});
