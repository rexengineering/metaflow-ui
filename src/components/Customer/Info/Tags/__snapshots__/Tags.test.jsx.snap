// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagsInfo /> should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="name"
  >
    <div
      class="makeStyles-header-7"
    >
      <h6
        class="MuiTypography-root makeStyles-label-8 MuiTypography-h6"
      >
        Tags
      </h6>
    </div>
    <div>
      <div
        class="makeStyles-container-9"
      >
        <p
          class="MuiTypography-root makeStyles-label-10 MuiTypography-body1"
        >
          Direct Buyer
        </p>
        <div
          class="makeStyles-actionElement-11"
        />
      </div>
      <div
        class="makeStyles-container-9"
      >
        <p
          class="MuiTypography-root makeStyles-label-10 MuiTypography-body1"
        >
          RHL
        </p>
        <div
          class="makeStyles-actionElement-11"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
